agent/structs/config_entry_discoverychain.go

```diff
@@ -807,6 +843,51 @@ func (e *ServiceResolverConfigEntry) Validate() error {
 		return fmt.Errorf("Bad ConnectTimeout '%s', must be >= 0", e.ConnectTimeout)
 	}
 
+	if e.LoadBalancer != nil && e.LoadBalancer.EnvoyConfig != nil {
```

Comment:
Maybe just pre-emptively should add a little bit of extra validation to enforce: that if you have a non-nil LoadBalancer struct that exactly-one of the two options is populated and valid, like:
```
if e.LoadBalancer != nil {
  switch {
  case e.LoadBalancer.EnvoyConfig != nil:
    ... what you have here already to verify a non-empty struct
  case e.LoadBalancer.OpaqueConfig != "":
    // totally fine, there's no more validation we can do
  default:
    return fmt.Errorf("LoadBalancer specified but not configured")
  }
}
```

Reply:
Resolving since `OpaqueConfig` was removed and `EnvoyConfig` was promoted